src/test/java/com/google/collegeplanner/SemesterSchedulerTest.java

```diff
@@ -0,0 +1,211 @@
+        new Meeting[] {TuThuLateMorning}));
+
+    scheduler = new SemesterScheduler(allClasses);
+
+    ArrayList<Schedule> expectedSchedules = new ArrayList<Schedule>();
+    expectedSchedules.add(new Schedule());
+
+    expectedSchedules.get(0).addClass(compSciClasses.get(0));
+    expectedSchedules.get(0).addClass(englishClasses.get(0));
+    expectedSchedules.get(0).addClass(mathClasses.get(0));
+    expectedSchedules.get(0).addClass(chemClasses.get(0));
+
+    Assert.assertEquals(expectedSchedules, scheduler.getPossibleSchedules());
+  }
+
+  @Test
+  public void multipleMeetingsPerSectionMultiplePossibleSchedules() {
+    compSciClasses.add(new Section("0CMSC101 (MWF Morning and Friday Afternoon)", "CMSC101", null,
+        10, 30, null, new Meeting[] {MWFMorning, fridayOnlyAfternoon}));
+
+    compSciClasses.add(new Section("1CMSC101 (TuThuLateAfternoon)", "CMSC101", null, 10, 30, null,
+        new Meeting[] {TuThuLateAfternoon}));
+
+    englishClasses.add(new Section("0ENGL101 (TuThu Morning, TuThuAfternoon)", "ENGL101", null, 10,
+        30, null, new Meeting[] {TuThuMorning, TuThuAfternoon}));
+
+    mathClasses.add(new Section("0MATH101 (MWF Late Morning)", "MATH101", null, 10, 30, null,
+        new Meeting[] {MWFLateMorning}));
+    mathClasses.add(new Section(
+        "1MATH101 (MWF Afternoon)", "MATH101", null, 10, 30, null, new Meeting[] {MWFAfternoon}));
+
+    chemClasses.add(new Section("0CHEM101 (TuThu Late Morning)", "CHEM101", null, 10, 30, null,
+        new Meeting[] {TuThuLateMorning}));
+
+    scheduler = new SemesterScheduler(allClasses);
+
+    ArrayList<Schedule> expectedSchedules = new ArrayList<Schedule>();
+    expectedSchedules.add(new Schedule());
+    expectedSchedules.add(new Schedule());
+    expectedSchedules.add(new Schedule());
+
+    expectedSchedules.get(0).addClass(compSciClasses.get(0));
+    expectedSchedules.get(0).addClass(englishClasses.get(0));
+    expectedSchedules.get(0).addClass(mathClasses.get(0));
+    expectedSchedules.get(0).addClass(chemClasses.get(0));
+
+    expectedSchedules.get(1).addClass(compSciClasses.get(1));
+    expectedSchedules.get(1).addClass(englishClasses.get(0));
+    expectedSchedules.get(1).addClass(mathClasses.get(1));
+    expectedSchedules.get(1).addClass(chemClasses.get(0));
+
+    expectedSchedules.get(2).addClass(compSciClasses.get(1));
+    expectedSchedules.get(2).addClass(englishClasses.get(0));
+    expectedSchedules.get(2).addClass(mathClasses.get(0));
+    expectedSchedules.get(2).addClass(chemClasses.get(0));
+
+    assertEqualsLists(expectedSchedules, scheduler.getPossibleSchedules());
+  }
+
+  /**
+   * This function checks if two lists are the same (order does not matter)
+   * @param expected the list of expected schedules
+   * @param actual the list of actual schedules
+   */
+  public void assertEqualsLists(ArrayList<Schedule> expected, ArrayList<Schedule> actual) {
```

Comment:
any chance we could use `containsExactlyElementsIn` instead of writing your own (or using some other open source alternative)? This is part of the `com.google.common.truth` package - though I'm not sure how difficult it is to pull in (https://github.com/google/truth).